### ISSUE TITLE
not starting fetchers after rebalance

### ISSUE THREAD

**Issue by divan-9** [confluentinc/confluent-kafka-dotnet#195]
Hello, 
Sometimes, not always but quite often consumer does not start fetching after rebalancing. 
The similar problem is described in https://github.com/edenhill/librdkafka/issues/1142 but I haven't found an answer there. I guess, that we work with library in a wrong way and misunderstood something. Any help will be appreciated.

- We have several consumers in single group each polls single topic. 
- We have to process messages in multiple workers for performace reason. It's made by calling 'Poll' method of consumer in dedicated thread and routing them to workers pool
- Offsets are beeing commited manually. At-least-once delivery guarantee is the important requirement. We do this by collecting each processed message in buffer and flushing this buffer once a 3 sec / or by reaching the maximum size / and in OnPartitionsRevoked event handler
- We're using Confluent.Kafka 0.9.5

### How we poll 
```
nativeConsumer.Subscribe(topicName);

pollingThread = new Thread(_ =>
{
    while (!token.IsCancellationRequested)
    {
        try
        {
            nativeConsumer.Poll(TimeSpan.FromSeconds(1));
        }
        catch (Exception ex)
        {
            Logger.Error(ex, "Exception on polling");
            completionSource.SetException(ex);
            throw;
        }
    }

    completionSource.SetResult(true);
});
pollingThread.Start();
```

### How we commit offsets. 

```
var topicPartitionOffset = new TopicPartitionOffset(
    tp: new TopicPartition(topic, partitionId),
    offset: offset);

var result = await this.nativeConsumer.CommitAsync(new[] { topicPartitionOffset });

if (result.Error.HasError)
{
    LogError(result.Error);
}
```

### Log line before stuck
```
consumer-1 [thrd:main]: Group "group-1": starting fetchers for 3 assigned partition(s) in join-state assigned (usable_offsets=no, v3, line 2243) FETCHSTART  
```

**Comment by treziac** [comment]
There shouldn't be any problem with the code. Your way to provision a Thread for polling and dispach to workers is the proper way.
Do you know if your code blocks on a line (ex CommitAsync or on a Poll) or just not fetching with no blocked thread (pollingThread or an other one)?

Can you also provide
- Your OS
- Your config
- Your rebalance callback (OnPartitionAssign / OnPartitionRevoked)

Thanks!

**Comment by MadRabbitKiller** [comment]
Thank you for reply

As far as I know code is never blocks (at least this code :). By the time the other instances start consume messages after rebalancing the failed one stops producing MessageReceived events and, what worse, holds assigned partitions  which leads to growing lag.

- Windows server 2012R2 x64
- Config
```
            var config = new Dictionary<string, object>
            {
                { "group.id", groupId },
                { "enable.auto.commit", false },
                { "bootstrap.servers", brokerList },
                { "statistics.interval.ms", 60000 },
                { "log.connection.close", false },
                { "heartbeat.interval.ms", 3000 },
                { "session.timeout.ms", 15000 }
            };
```
- OnPartitionsAssign 
```
       private void OnPartitionsAssigned(object sender, List<TopicPartition> e)
        {
            try
            {
                var partitionIds = e.Select(x => x.Partition);
                Logger.Info($"Assigned partitions: {partitionIds.Dump()}");
                this.nativeConsumer.Assign(e);
            }
            catch (Exception ex)
            {
                Logger.Error(ex, $"Exception on partition assignment");
            }
        }
```
-  OnPartitionsRevoked is not used. (yes, I mentioned about commiting offsets on this event, but it was not actually implemented yet)

**Comment by andreycha** [comment]
@divan-9  this one might fix your issue, keep an eye on the next release: https://github.com/edenhill/librdkafka/pull/1621

**Comment by mhowlett** [comment]
this is a librdkafka issue, and I suspect it's resolved in 0.11.5 (and 1.0-experimental), given https://github.com/edenhill/librdkafka/pull/1621 is merged, but I'm leaving this open and marking it `investigate further` since i think it's a high priority issue.

**Comment by anchitj** [comment]
Closing as the fix is merged.
